### PR TITLE
Created basic unit tests using Vitest

### PR DESCRIPTION
**Pull request by gang21**
Currently there are 6 testing suites with 22 total tests that test whether basic components exist in each of the project pages. 

To run the testing suite, run the following commands:
```
cd front
npm test
```

The test suite should also run after every push/pull request

### CODE REVIEW

**Comment by ribru17** [comment]
Nice!! thanks :)

Maybe it would be better if this branch was rebased off of my vitest one? I based that one off a few commits back so it doesn't have any of the extra babel/jest dependencies, I think this branch still has all that so our `node_modules` will be larger than needed

Also this is a nitpick, but in the future if you need to fix up some things you can also just `git commit -a --amend --no-edit` which just edits the most recent commit, so you don't have to add a lot of commits for each tiny fixup. [I usually alias this to just `git amend`](https://github.com/ribru17/.dotfiles/blob/39746059b500358ee35095e8f04b3d95d74d78a2/.config/git/config#L51)

File: front/package.json [hunk 1]
```diff
@@ -11,17 +11,12 @@
...
-    "@testing-library/user-event": "^13.5.0",
     "@types/jest": "^27.5.2",
-    "@types/node": "^16.18.113",
-    "@types/react": "^18.3.11",
-    "@types/react-dom": "^18.3.0",
     "axios": "^1.7.7",
     "formik": "^2.4.6",
     "moment": "^2.30.1",
     "notistack": "^3.0.1",
+    "prettier": "^3.3.3",
```
Comment (ribru17) [inline front/package.json:19]: I think this should be in `devDependencies` rather than regular `dependencies`

**Comment by ribru17** [comment]
Hey @gang21 I have created #66 just to set up vitests (which is rebased from a previous commit without babel/jest etc and thus doesn't have all of the bloat). I think it is easiest to merge that and then maybe you can cherry pick your tests onto that commit? Does that sound ok with you or do you think there is a better way to go about things?

**Comment by ribru17** [comment]
@gang21 I have pushed up to this branch removing old dependencies (check diffstat! :tada: ) and run formatting / fixed the tester CI, all that is left now is to fix the tests. I was going to leave this to you since you know more about how to do this than me :) Looks like all tests pass except for two on resourcepage, then we can merge. Thanks for your patience!!

**Comment by ribru17** [comment]
@gang21 How do you feel about commenting out the faililng resource page tests just to get CI green then we can fix them as a separate PR?

**Comment by gang21** [comment]
I agree, I commented out the failing tests for the resource page for now and will look into why that isn't working. Sorry for the inconvenience all!

**Comment by gang21** [comment]
I'm wondering if the tests are still failing because the fixed test case hasn't been merged in yet. Anyone have thoughts on this?

**Comment by ribru17** [comment]
Right now it is failing because all `package-lock.json` files were removed. The commit squashes I applied yesterday basically fixed up everything but they seem to have been undone now :thinking: Did you have to force push these changes? If not I am not sure how my other commit was overridden. I can take a look soon I have a quiz at 4 sadly lol.

**Comment by gang21** [comment]
ack yea i might've done something when trying to merge changes sorry!

**Comment by ribru17** [comment]
Nah no worries! Are you free to hop in a short call later tn maybe? Maybe some time after 7 thats when I get home

**Comment by gang21** [comment]
yes!

**Pull request merged**